src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultFileCredentialImpl.java

```diff
@@ -0,0 +1,128 @@
+package com.datapipe.jenkins.vault.credentials.common;
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.UUID;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+
+import static com.datapipe.jenkins.vault.configuration.VaultConfiguration.engineVersions;
+import static com.datapipe.jenkins.vault.credentials.common.VaultHelper.getVaultSecret;
+import static com.datapipe.jenkins.vault.credentials.common.VaultHelper.getVaultSecretKey;
+
+
+
+public class VaultFileCredentialImpl extends AbstractVaultBaseStandardCredentials implements VaultFileCredential {
+
+    private static final long serialVersionUID = 1L;
+
+    private String fileName;
+    private Boolean useKey;
+    private String vaultKey;
+
+    @DataBoundConstructor
+    public VaultFileCredentialImpl(CredentialsScope scope, String id,
+        String description) {
+        super(scope, id, description);
+        this.fileName = UUID.randomUUID().toString();
+    }
+
+    @NonNull
+    public String getVaultKey() {
+        return vaultKey;
+    }
+
+    @DataBoundSetter
+    public void setVaultKey(String vaultKey) {
+        this.vaultKey = vaultKey;
+    }
+
+    @NonNull
+    public Boolean getUseKey() {
+        return useKey;
+    }
+
+    @DataBoundSetter
+    public void setUseKey(Boolean useKey) {
+        this.useKey = useKey;
+    }
+
+
+    @NonNull
+    public String getFileName() {
+        return fileName;
+    }
+
+    @NonNull
+    @Override
+    public InputStream getContent() {
+        String content;
+        if (useKey != null && useKey) {
+            content = getVaultSecretKeyValue(vaultKey);
+        } else {
+            Map<String, String> s = getVaultSecretValue();
+            content = new JSONObject().fromObject(s).toString();
```

Comment:
A static method exist, no need to call new :) https://github.com/jenkinsci/json-lib/blob/16052d1f74908fb59430eddd73f42aec2469b313/src/main/jdk15/net/sf/json/JSONObject.java#L138-L140
```suggestion
            content = JSONObject.fromObject(s).toString();
```